napari/_qt/_tests/test_qt_notifications.py

```diff
@@ -64,6 +68,25 @@ def store_widget(self, *args, **kwargs):
     monkeypatch.setattr(_QtMainWindow, "current", mock_current_main_window)
 
 
+@dataclass
+class ShowStatus:
+    show_count: int = 0
+
+
+@pytest.fixture
+def show_dialog_mock(monkeypatch, qtbot):
+
+    stat = ShowStatus()
+
+    def mock_show_dialog(self):
+        stat.show_count += 1
+        qtbot.add_widget(self)
```

Comment:
This only adds the widget on each call of `show`. If there are no calls that means it won't be added, which I guess is why we also have the other calls to `add_widget` in the tests. Could we patch `__init__` so that we add it exactly once instead?

Reply:
could you check now?

Reply:
If I recall correctly, all the tests were passing before this change, but now the flaky windows 3.8 pyside2 test is failing again due to `test_qt_notifications`. We might have just got lucky before this change and I don't see any reason why the new changes are worse than the previous ones, but just wondered if you had any thoughts?

Reply:
I found 2 places.. 
But they are failing only if run all tests, not when running individual tests.